presto-main/src/test/java/com/facebook/presto/util/TestGraphvizPrinter.java

```diff
@@ -0,0 +1,141 @@
+            return TEST_CONNECTOR_TABLE_HANDLE_STR;
+        }
+    };
+
+    @Test
+    public void testPrintLogical()
+    {
+        String actualStr = GraphvizPrinter.printLogical(
+                ImmutableList.of(createTestPlanFragment(0, createTestTableScanNode())),
+                TestingSession.testSessionBuilder().build(),
+                FunctionAndTypeManager.createTestFunctionAndTypeManager());
+        String expectStr = String.join(
+                "\n",
+                new String[] {"digraph logical_plan {",
+                        "subgraph cluster_0 {",
+                        "label = \"SOURCE\"",
+                        String.format("plannode_1[%s];", TEST_TABLE_SCAN_NODE_ATTR_STR),
+                        "}",
+                        "}",
+                        ""});
+        assertEquals(actualStr, expectStr);
+    }
+
+    @Test
+    public void testPrintDistributed()
+    {
+        SubPlan tableScanNodeSubPlan = new SubPlan(
+                createTestPlanFragment(0, createTestTableScanNode()),
+                ImmutableList.of());
+        SubPlan nestedSubPlan = new SubPlan(
+                createTestPlanFragment(1, createTestTableScanNode()),
+                ImmutableList.of(tableScanNodeSubPlan));
+        String actualNestedSubPlanStr = GraphvizPrinter.printDistributed(
+                nestedSubPlan,
+                TestingSession.testSessionBuilder().build(),
+                FunctionAndTypeManager.createTestFunctionAndTypeManager());
+        String expectNestedSubPlanStr = String.join(
+                "\n",
+                new String[] {"digraph distributed_plan {",
+                        "subgraph cluster_1 {",
+                        "label = \"SOURCE\"",
+                        String.format("plannode_1[%s];", TEST_TABLE_SCAN_NODE_ATTR_STR),
+                        "}",
+                        "subgraph cluster_0 {",
+                        "label = \"SOURCE\"",
+                        String.format("plannode_2[%s];", TEST_TABLE_SCAN_NODE_ATTR_STR),
+                        "}",
+                        "}",
+                        ""});
+
+        assertEquals(actualNestedSubPlanStr, expectNestedSubPlanStr);
+    }
+
+    // bare-minimum PlanFragment instance for testing
+    private static PlanFragment createTestPlanFragment(int id, PlanNode node)
+    {
+        return new PlanFragment(
+                new PlanFragmentId(id),
+                node,
+                ImmutableSet.of(),
+                SOURCE_DISTRIBUTION,
+                ImmutableList.of(TEST_TABLE_SCAN_NODE_ID),
+                new PartitioningScheme(Partitioning.create(SINGLE_DISTRIBUTION, ImmutableList.of()), ImmutableList.of()),
+                StageExecutionDescriptor.ungroupedExecution(),
+                false,
+                StatsAndCosts.empty(),
+                Optional.empty());
+    }
+
+    // bare-minimum PlanNode instance for testing
+    private static PlanNode createTestTableScanNode()
```

Comment:
Thanks for pointing that out, I missed that class.
After looking into it, I think using that class would save us 2 lines and provides better SPoC, but it requires constructing a `MetaData` object, which I can't seem to find an easy way to do (it would take up more lines).